### DISCUSSION TITLE
Locations Import

### DISCUSSION THREAD

**Discussion by heidl89** [netbox-community/netbox#16856] [Help Wanted!]
Hi. 
I am struggeling with the CSV Import of our Locations. 

Our documentation/CSV looks like this: 

1: 
Main Station
*First Floor
**Technic Room 1.2

2: 
Main Station South
*First Floor
**Technic Room 2.5

The import of the Technic Rooms wont work, because of non unique "First Floor". 
Does anybody know how to fix this?

TY
Greets Markus

**Comment by jsenecal** [comment]
Isn't that related to the slug field being non-unique ?

**Comment by heidl89** [comment]
Unfortunately it says parent name...

![import_locations](https://github.com/netbox-community/netbox/assets/131429239/6a8a724a-b44c-44c0-a6c5-3f4de0e0c2ab)

**Reply by jsenecal** [reply]
I don't see that in your screenshot

**Reply by candlerb** [reply]
Last line shows Field: parent, Accessor: name.

**Comment by candlerb** [comment]
It would be helpful if you showed the exact CSV you're trying to import.

Try using "parent.slug" instead of "parent" as the column name.  Make sure each location that you create has a unique slug, like "main-station-first-floor" and "main-station-south-first-floor", so that it can be referenced by a child location.

The following works for me in Netbox 4.0.7:

```
# Sites
name,slug,status
Main Station,main-station,planned
Main Station South,main-station-south,planned

# Locations
site,name,slug,status,parent.slug
Main Station,First Floor,main-station-first-floor,planned,
Main Station,Technic Room 1.2,main-station-first-floor-technic-room-1-2,planned,main-station-first-floor
Main Station South,First Floor,main-station-south-first-floor,planned,
Main Station South,Technic Room 2.5,main-station-south-first-floor-technic-room-2-5,planned,main-station-south-first-floor
````

![image](https://github.com/netbox-community/netbox/assets/44789/ac68c9e3-2758-4167-8956-6b100a727655)

You should also be able to use "parent.id" if you know the database ID of the parent location (but this won't be known until the parent has been imported)

Admittedly this is not well [documented](https://netboxlabs.com/docs/netbox/en/stable/getting-started/populating-data/#bulk-import-csvyaml) or tested behaviour of CSV upload.

**Reply by heidl89** [reply]
That works!
Thank you very much :)

**Comment by heidl89** [comment]
the import with parent.slug in the csv header works
